mycli/sqlcompleter.py

```diff
@@ -372,45 +372,55 @@ def find_matches(text, collection, start_only=False, fuzzy=True, casing=None):
         last = last_word(text, include='most_punctuations')
         text = last.lower()
 
-        completions = []
+        if casing == 'auto':
+            casing = 'lower' if last and last[-1].islower() else 'upper'
+
+        def apply_case(kw):
+            if casing is None:
+                return kw
+            if casing == 'upper':
+                return kw.upper()
+            return kw.lower()
+
+        matches = []
 
         if fuzzy:
             regex = '.*?'.join(map(escape, text))
             pat = compile('(%s)' % regex)
             for item in collection:
                 r = pat.search(item.lower())
                 if r:
-                    completions.append((len(r.group()), r.start(), item))
+                    matches.append(
+                        (len(r.group()), r.start(), apply_case(item)))
         else:
             match_end_limit = len(text) if start_only else None
             for item in collection:
                 match_point = item.lower().find(text, 0, match_end_limit)
                 if match_point >= 0:
-                    completions.append((len(text), match_point, item))
-
-        if casing == 'auto':
-            casing = 'lower' if last and last[-1].islower() else 'upper'
+                    matches.append((len(text), match_point, apply_case(item)))
 
-        def apply_case(kw):
-            if casing == 'upper':
-                return kw.upper()
-            return kw.lower()
-
-        return (Completion(z if casing is None else apply_case(z), -len(text))
-                for x, y, z in completions)
+        return matches
 
     def get_completions(self, document, complete_event, smart_completion=None):
         word_before_cursor = document.get_word_before_cursor(WORD=True)
+
+        def sorted_completions(matches):
+            # sort by match point, then match length, then item text
+            matches = sorted(matches, key=lambda m: (m[1], m[0], m[2].lower()))
```

Comment:
As long as the sort is casefolding it might as well also do

```
m[2].lower().strip('`')
```

Reply:
Yeah, I can certainly add that. Pretty sure I've seen someone else complain in another issue about the back tick messing with sorting.